src/plugins/kibana/public/settings/sections/indices/add_data_steps/pipeline_setup/lib/processor.js

```diff
@@ -0,0 +1,25 @@
+import _ from 'lodash';
+
+export default function Processor(processorType) {
```

Comment:
I'm not generally a huge fan of them, but this might actually be a place where ES6 classes are a natural fit. Processor could be the super class and each processor type a sub-class. That said, if you're not into the idea, I would be fine with what you have here.